protocols/imap/src/main/java/org/apache/james/imap/encode/ESearchResponseEncoder.java

```diff
@@ -66,10 +63,14 @@ protected void doEncode(ImapMessage acceptableMessage, ImapResponseComposer comp
         if (options.contains(SearchResultOption.COUNT)) {
             composer.message(SearchResultOption.COUNT.name()).message(count);
         }
-        if (all != null && all.length > 0 && options.contains(SearchResultOption.ALL)) {
+        if (!useUid && all != null && all.length > 0 && options.contains(SearchResultOption.ALL)) {
```

Comment:
I do not uderstand why a hudge part of the changes in imap subpart was needed. From what I see, UIDs can now be Optional, and there is also a type change from IdRange to UidRange... I found hard to review the IMAP changes, and it is not clear for me, why it is related to maibox's MessageUid....